### PR TITLE
Creating a new collection view controller using stream 

### PR DESCRIPTION
**Pull request by MartinJibber**

### CODE REVIEW

File: iOS/App/Lorem.swift [hunk 1]
```diff
@@ -81,56 +81,6 @@ class Lorem {
                             handle: String())
     }
 
-    class func systemConversation() -> SystemConversation {
```
Comment (MartinJibber) [inline iOS/App/Lorem.swift:84]: We no longer need system conversations because Stream makes it easy to mock data. I've deleted all references to it.

File: iOS/Flows/Conversation/ConversationCoordinator.swift [hunk 2]
```diff
@@ -14,7 +14,7 @@ import Combine
 
 class ConversationCoordinator: PresentableCoordinator<Void> {
 
-    lazy var conversationVC = ConversationViewController(conversation: self.conversation, delegate: self)
+    lazy var conversationVC = new_ConversationViewController(conversation: self.conversation)
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/ConversationCoordinator.swift:17]: Presenting the new conversation VC now. The old VC still exists and you can't use it if needed.

File: iOS/Flows/Conversation/ConversationCoordinator.swift [hunk 3]
```diff
@@ -25,11 +25,11 @@ class ConversationCoordinator: PresentableCoordinator<Void> {
     }()
     private var cancellables = Set<AnyCancellable>()
 
-    var conversation: DisplayableConversation?
+    var conversation: Conversation?
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/ConversationCoordinator.swift:28]: DisplayableConvesation and ConversationType are both gone. We now directly reference Stream ChatChannels, however I've typealiased it so that we use the "Conversation" semantics.

File: iOS/Flows/Conversation/ConversationViewController+Messaging.swift [hunk 4]
```diff
@@ -34,106 +34,61 @@ extension ConversationViewController: SwipeableInputAccessoryViewDelegate {
         }
     }
 
-    func load(activeConversation: DisplayableConversation) {
-        switch activeConversation.conversationType {
-        case .system(_):
-            break
-        case .conversation:
-            self.loadMessages(for: activeConversation.conversationType)
-        }
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/ConversationViewController+Messaging.swift:43]: Conversation type is gone, so no more need for these switches.

Comment (MartinJibber) [reply iOS/Flows/Conversation/ConversationViewController+Messaging.swift:43]: No functional changes to this file. Just getting it to compile.

File: iOS/Flows/Conversation/ConversationViewController+Updates.swift [hunk 5]
```diff
@@ -11,67 +11,49 @@ import StreamChat
 
 extension ConversationViewController {
     
-    func loadMessages(for conversationType: ConversationType) {
+    func loadMessages(for conversation: Conversation) {
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/ConversationViewController+Updates.swift:14]: No functional changes.

File: iOS/Flows/Conversation/ConversationViewController.swift [hunk 6]
```diff
@@ -22,11 +22,12 @@ protocol ConversationViewControllerDelegate: AnyObject {
 class ConversationViewController: FullScreenViewController, CollectionViewInputHandler {
 
     let blurView = UIVisualEffectView(effect: UIBlurEffect(style: .dark))
-    lazy var detailVC = ConversationDetailViewController(conversation: self.conversation, delegate: self.delegate)
+    lazy var detailVC = ConversationDetailViewController(conversation: self.conversation,
+                                                         delegate: self.delegate)
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/ConversationViewController.swift:26]: No functional changes

File: iOS/Flows/Conversation/Manager/ConversationCollectionViewManager+Extensions.swift [hunk 7]
```diff
@@ -12,26 +12,26 @@ import Combine
...
-
-        let copyWithConsumers = try await messageCopy.updateConsumers(with: current)
-        await self.updateItem(with: copyWithConsumers)
-        if message.context == .timeSensitive {
-            UIApplication.shared.applicationIconBadgeNumber = 0
-        }
-
-        //call update on the actual message and update on callback
-        try await message.updateConsumers(with: current)
+        #warning("Replace")
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Manager/ConversationCollectionViewManager+Extensions.swift:15]: No changes

File: iOS/Flows/Conversation/Manager/ConversationSectionable.swift [hunk 8]
```diff
@@ -12,7 +12,7 @@ class ConversationSectionable {
 
     var date: Date
     var items: [Messageable] = []
-    var conversationType: ConversationType?
+    var conversation: Conversation?
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Manager/ConversationSectionable.swift:15]: Lots of simple type name changes below. Keep going until another comment.

File: iOS/Flows/Home/_New/ConversationCollectionViewLayout.swift [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+//  new_ConversationCollectionViewLayout.swift
+//  new_ConversationCollectionViewLayout
+//
+//  Created by Martin Young on 9/15/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+
+class ConversationCollectionViewLayout: UICollectionViewCompositionalLayout {
```
Comment (MartinJibber) [inline iOS/Flows/Home/_New/ConversationCollectionViewLayout.swift:11]: OK, here's the new stuff. Just the beginnings of the conversation layout. Basically just makes a horizontal list of messages. Pretty ugly for now but we'll come back to this.

File: iOS/Flows/Home/_New/ConversationViewControllerDataSource.swift [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+
+typealias ConversationCollectionSection = ConversationCollectionViewDataSource.SectionType
+typealias ConversationCollectionItem = ConversationCollectionViewDataSource.ItemType
+
+class ConversationCollectionViewDataSource: CollectionViewDataSource<ConversationCollectionViewDataSource.SectionType,
```
Comment (MartinJibber) [inline iOS/Flows/Home/_New/ConversationViewControllerDataSource.swift:15]: The conversation data source. A subclass of our base DataSource type.

This particular interaction doesn't support threading, but I finally figured out how to do it right and it'll be in another PR.

File: iOS/Flows/Home/_New/ConversationViewControllerDataSource.swift [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+
+    enum SectionType: Hashable {
+        case basic(ChannelId)
+    }
+
+    enum ItemType: Hashable {
+        case message(MessageId)
+    }
+
+    var handleDeleteMessage: ((ChatMessage) -> Void)?
```
Comment (MartinJibber) [inline iOS/Flows/Home/_New/ConversationViewControllerDataSource.swift:26]: Callbacks for when the ContextMenu tells us to delete a message.

File: iOS/Flows/Home/_New/ConversationViewControllerDataSource.swift [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+
+    private let contextMenuDelegate: ContextMenuInteractionDelegate
+
+    override init(collectionView: UICollectionView) {
+        self.contextMenuDelegate = ContextMenuInteractionDelegate(collectionView: collectionView)
```
Comment (MartinJibber) [inline iOS/Flows/Home/_New/ConversationViewControllerDataSource.swift:31]: Annoyingly the delegate ContextMenuDelegate needs to be an NSObject so I made a proxy object that inherits from NSObject. Alternatively I could make all datasource objects NSObject.

File: iOS/App/Lorem.swift (hunk 1 above)
Comment (Benjamindodgson) [reply iOS/App/Lorem.swift:84]: sick

**Review by Benjamindodgson — APPROVED** [review]
